lib/internal/webstreams/readablestream.js

```diff
@@ -488,11 +488,11 @@ class ReadableStream {
         const result = readableStreamReaderGenericCancel(reader, value);
         readableStreamReaderGenericRelease(reader);
         await result;
-        return { done: true, value };
+        return { done: true, value }; // eslint-disable-line node-core/avoid-prototype-pollution
```

Comment:
instead of disabling the rule, should we add an own non-callable then?

Reply:
I've considered it, but it seems the ES spec doesn't (https://tc39.es/ecma262/#sec-createiterresultobject), so it probably makes the most sense if it behaves as native async iterators.

Reply:
Sure it doesn't, but the ES spec also doesn't care about preventing prototype pollution.

I suppose I see the argument that if this iterator result is passed to users, that it might as well match their expectations and *be* vulnerable to pollution, but tbh that's kind of an argument against this PR, since anyone providing a callable Object.prototype.then is going to wreak a special kind of havoc on any program.

Reply:
This API (`globalThis.ReadableStream`) is also present on browsers, but unfortunately none has implemented the async iterator part of it. Deno has implemented it, and it looks like they are protecting against this kind of prototype pollution: https://github.com/denoland/deno/blob/e6e28981909f220ff0b98a13c692c0203eaf6035/ext/web/06_streams.js#L4381-L4393
Given that, I think it makes sense to align with them, rather than the ES spec.

Reply:
Hum it looks like the WPT forbids to do any sort of protection here 🤔
https://github.com/nodejs/node/blob/6dde81042f0357cf465b4075790e9e35c9a993c5/test/fixtures/wpt/streams/readable-streams/async-iterator.any.js#L9-L16

Reply:
so deno's willfully violating these?

Reply:
This test doesn't pass for them, yes: https://github.com/denoland/deno/blob/d0e78ca5c6abb75b306e669ed844dd3705992677/tools/wpt/expectation.json#L1316